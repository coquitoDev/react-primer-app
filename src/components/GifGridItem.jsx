import PropTypes from 'prop-types';



export const GifGridItem = ({ title, url }) => {

     

     return (
          <div className="card">
               <img src={ url } alt={ title } />
               <p>{ title }</p>

          </div>
     )
}

GifGridItem.propTypes = {
     url: PropTypes.string.isRequired,
     title: PropTypes.string.isRequired, 
}
