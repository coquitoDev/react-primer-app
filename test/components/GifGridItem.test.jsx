import { render, screen,  } from '@testing-library/react';
import { GifGridItem } from '../../src/components/GifGridItem';


describe('Prueba en <GifGridItem />', () => { 

     const title = 'Saitama';
     const url   = 'https://one-punch.com/saitama.jpg'

     test('Debe hacer match con el snapshot Renderizar titulo y cargar url', () => {

          const { container } = render( <GifGridItem title={ title } url={ url } />)
          expect( container ).toMatchSnapshot();

     });

     test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
          render( <GifGridItem title={ title } url={ url } />) ;
          // screen.debug();
          console.log(screen.getByRole("img"))

     });

})