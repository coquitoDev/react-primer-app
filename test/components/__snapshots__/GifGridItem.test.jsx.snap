// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGridItem /> Debe hacer match con el snapshot Renderizar titulo y cargar url 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
